File: backend/node/src/domain/modules/users/interfaces/user-repository.interface.ts
import type {
	IBaseReadRepository,
	IBaseRepository,
} from '../../../base/interfaces/base-repository.interface';
import type { UserEntity } from '../entities/user.entity';

export interface IUserReadRepository extends IBaseReadRepository<UserEntity> {
	findOneByEmail: (email: string, id?: string) => Promise<UserEntity | null>;
}

export interface IUserRepository
	extends IBaseRepository<UserEntity>,
		Omit<IUserReadRepository, 'findOneByEmail'> {}

File: backend/go/domain/modules/users/user-repository.interface.go
package users

import (
	"github.com/heyhusen/playground/backend/go/domain/base"
)

type IUserReadRepository interface {
	base.IBaseReadRepository[UserEntity]

	FindOneByEmail(email string, id *string) (UserEntity, error)
}

type IUserRepository interface {
	base.IBaseReadRepository[UserEntity]
	base.IBaseRepository[UserEntity]
}
